### DISCUSSION TITLE
Should we be using the new `authStateReady()` method instead of `currentUser`?

### DISCUSSION THREAD

**Discussion by dosstx** [vuejs/vuefire#1399] [Questions and Help] [answered]
Firebase Auth now has a `authStateReady()` method. Anyone seen this and is it worth replacing `currentUser` with it? Is it more performant? 

> Implemented authStateReady(), which returns a promise that resolves immediately when the initial auth state is settled and currentUser is available. When the promise is resolved, currentUser might be a valid user or null if there is no user signed in currently.

Source: https://firebase.google.com/support/release-notes/js#version_1010_-_july_20_2023

**Comment by posva** [comment, marked as answer]
I suppose you mean `getCurrentUser()`, not `currentUser`. In that case, it shouldn't make a difference.
